common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -396,7 +396,9 @@ public boolean startsWith(final UTF8String prefix, int collationId) {
     if (collationId == CollationFactory.LOWERCASE_COLLATION_ID) {
       return this.toLowerCase().startsWith(prefix.toLowerCase());
     }
-    return matchAt(prefix, 0, collationId);
```

Comment:
if there's code in `UTF8String` that's no longer used, we should delete it (for example, the overloaded `matchAt`)

Reply:
Deleted